cmake/engine_git.cmake

```diff
@@ -1,50 +1,83 @@
-function (set_git_version BRANCH HASH VERSION)
+function (set_git_version BRANCH HASH VERSION_UNFORMATTED)
     set(OBENGINE_GIT_BRANCH ${BRANCH} CACHE STRING "ObEngine git branch" FORCE)
     set(OBENGINE_GIT_COMMIT_HASH ${HASH} CACHE STRING "ObEngine git commit hash" FORCE)
-    set(OBENGINE_VERSION ${VERSION} CACHE STRING "ObEngine version" FORCE)
+
+    string(REGEX MATCH "v(.+)\\.(.+)\\.(.+)$" VERSION "${VERSION_UNFORMATTED}")
+
+    set(OBENGINE_VERSION ${VERSION} CACHE STRING "ObEngine version string" FORCE)
+    set(VERSION_MAJOR ${CMAKE_MATCH_1})
+    set(VERSION_MINOR ${CMAKE_MATCH_2})
+    set(VERSION_PATCH ${CMAKE_MATCH_3})
+
+    string(REGEX MATCH "([0-9]+)$" RES ${VERSION_MAJOR})
```

Comment:
The cause of the CI fail is right here :+1:
```
CMake Error at /home/runner/work/ObEngine/ObEngine/cmake/engine_git.cmake:12 (string):
  string sub-command REGEX, mode MATCH needs at least 5 arguments total to
  command.
Call Stack (most recent call first):
  /home/runner/work/ObEngine/ObEngine/cmake/engine_git.cmake:78 (set_git_version)
  /home/runner/work/ObEngine/ObEngine/cmake/engine_git.cmake:93 (configure_obengine_git)
src/Core/CMakeFiles/ConfigureObEngineGit.dir/build.make:74: recipe for target 'src/Core/CMakeFiles/ConfigureObEngineGit' failed

CMakeFiles/Makefile2:746: recipe for target 'src/Core/CMakeFiles/ConfigureObEngineGit.dir/all' failed
```

Reply:
Just a hint but that might happen when the `git` executable is not found, thus not matching the previous regex ?

Reply:
I'll look into it.

Reply:
It turned out that tags have not been fetched.